src/django/api/views/facility/facility_list_view_set.py

```diff
@@ -102,6 +121,44 @@ def _extract_header_rows(self, file, request):
         rows = map(clean_row, rows)
         return header, rows
 
+    @staticmethod
+    def __get_serializer(
+            file: Union[InMemoryUploadedFile, TemporaryUploadedFile]
+            ) -> ContriCleanerSerializer:
+        sector_split_pattern = r', |,|\|'
+        ext = file.name[-4:]
+        if ext == 'xlsx':
```

Comment:
Thanks for the comment.

1. I have used the `os` Python module, which is handier than the previous solution.
2. Regarding .xls, I agree with you. However, to preserve backward compatibility of the functionality, I wouldn't include such changes in this PR. I think it can be done as a separate task.